### PR TITLE
Radio buttons!

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

(Started out with a single shared `radio-button.js`, split it out because the style for native is pretty different, different colors, no hover effect.)

### CODE REVIEW

File: shared/common-adapters/checkbox.stories.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const load = () => {
       <Checkbox {...commonProps} label="Checkbox Unchecked Enabled" checked={false} />
       <Checkbox {...commonProps} label="Checkbox Checked Enabled" checked={true} />
       <Checkbox {...commonProps} label="Checkbox Unchecked Disabled" checked={false} disabled={true} />
-      <Checkbox {...commonProps} label="Checkbox Checked Disabled" checked={true} />
+      <Checkbox {...commonProps} label="Checkbox Checked Disabled" checked={true} disabled={true} />
```
Comment (cjb) [inline shared/common-adapters/checkbox.stories.js]: (Unrelated bug.)

**Comment by keybase-ci-visdiff** [comment]
The commits 400111d7deb4...19959f66faf3 introduce visual changes on linux. <details><summary>:mag_right: 4 new</summary>

 * Added: **Radio_Button-Normal_unchecked**  
   Radio_Button-Normal_unchecked rendered
 * Added: **Radio_Button-Normal_selected**  
   Radio_Button-Normal_selected rendered
 * Added: **Radio_Button-Disabled_unselected**  
   Radio_Button-Disabled_unselected rendered
 * Added: **Radio_Button-Disabled_selected**  
   Radio_Button-Disabled_selected rendered
</summary>

File: shared/common-adapters/radio-button.desktop.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// @flow
+import React from 'react'
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: once my pr is in this'll likely change to import * as React
and the React.Component<Props, {hovered: boolean}> etc

File: shared/common-adapters/radio-button.desktop.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (this.props.selected || (this.state.hovered && !this.props.disabled)) {
+      borderColor = globalColors.blue
+    }
+
+    const boxStyle = {
+      ...transition('background'),
+      backgroundColor: this.props.selected ? globalColors.blue : 'inherit',
+      border: `solid 1px ${borderColor}`,
+      borderRadius: 100,
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: likely want this to be 100%

File: shared/common-adapters/radio-button.desktop.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  render() {
+    let borderColor = globalColors.black_10
+
+    if (this.props.selected || (this.state.hovered && !this.props.disabled)) {
+      borderColor = globalColors.blue
+    }
+
+    const boxStyle = {
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: we should use glamorous here and not use onMouseEnter/leave

File: shared/common-adapters/radio-button.desktop.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      opacity: this.props.disabled ? 0.4 : 1,
+      position: 'relative',
+      width: RADIOBUTTON_SIZE,
+    }
+
+    const clickableStyle = this.props.disabled ? {} : globalStyles.clickable
+
+    return (
+      <div
+        style={{...styleContainer, ...clickableStyle, ...this.props.style}}
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: clickableStyle should just be in styleContainer

File: shared/common-adapters/radio-button.desktop.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        onClick={this.props.disabled ? undefined : () => this.props.onSelect(!this.props.selected)}
+        onMouseEnter={() => this.setState({hovered: true})}
+        onMouseLeave={() => this.setState({hovered: false})}
+      >
+        <div style={boxStyle}>
+          <div style={styleIcon} />
+        </div>
+        <Text type="Body" small={true} style={{color: globalColors.black_75}}>{this.props.label}</Text>
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: what is small={true}, not a real prop?

File: shared/common-adapters/radio-button.desktop.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+const styleContainer = {
+  ...globalStyles.flexBoxRow,
+  alignItems: 'center',
+}
+
+const styleIcon = {
+  ...transition('opacity'),
+  border: `solid 3px ${globalColors.white}`,
+  borderRadius: 10,
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: should be '100%'

File: shared/common-adapters/radio-button.desktop.js (hunk 5 above)
Comment (cjb) [inline shared/common-adapters/radio-button.desktop.js]: Hm, how would that work? clickableStyle depends on a prop (`disabled`).

**Comment by keybase-ci-visdiff** [comment]
The commits 13d4f1e08f14...a9a1f1adb578 introduce visual changes on linux. <details><summary>:mag_right: 4 new</summary>

 * Added: **Radio_Button-Normal_unchecked**  
   Radio_Button-Normal_unchecked rendered
 * Added: **Radio_Button-Normal_selected**  
   Radio_Button-Normal_selected rendered
 * Added: **Radio_Button-Disabled_unselected**  
   Radio_Button-Disabled_unselected rendered
 * Added: **Radio_Button-Disabled_selected**  
   Radio_Button-Disabled_selected rendered
</summary>

**Comment by chrisnojima** [comment]
Oh didn't see that that was a local var. it'll be defined in glamorous
anyways

**Comment by cjb** [comment]
@chrisnojima Switched to glamorous and rebased against your changes on master, ready for another look now!

**Comment by keybase-ci-visdiff** [comment]
The commits 13d4f1e08f14...d03ebe2619b0 introduce visual changes on linux. <details><summary>:mag_right: 4 new</summary>

 * Added: **Radio_Button-Normal_unchecked**  
   Radio_Button-Normal_unchecked rendered
 * Added: **Radio_Button-Normal_selected**  
   Radio_Button-Normal_selected rendered
 * Added: **Radio_Button-Disabled_unselected**  
   Radio_Button-Disabled_unselected rendered
 * Added: **Radio_Button-Disabled_selected**  
   Radio_Button-Disabled_selected rendered
</summary>

File: shared/common-adapters/radio-button.desktop.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+const RadioButton = ({disabled, label, onSelect, selected, style}: Props) => {
+  const clickableStyle = disabled ? {} : globalStyles.clickable
+
+  const StyledDiv = glamorous(glamorous.Div)(({border, disabled, selected}) => ({
+    '&:hover': {border: (selected || !disabled) && `solid 1px ${globalColors.blue}`},
+    border: `solid 1px ${globalColors.black_10}`,
+  }))
+
+  StyledDiv.defaultProps = {
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: likely better to just move this into the glamorous call above, you can do this

```
const StyledDiv = glamorous(glamorous.Div)({
  ...transition('background'),
  backgroundColor: selected ? globalColors.blue : 'inherit',
  borderRadius: '100%',
// ...
},
({border, disabled, selected}) => ({
     '&:hover': {border: (selected || !disabled) && `solid 1px ${globalColors.blue}`},
     border: `solid 1px ${globalColors.black_10}`,
   }))
```

File: shared/common-adapters/radio-button.desktop.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import type {Props} from './radio-button'
+import {globalStyles, globalColors, transition} from '../styles'
+
+export const RADIOBUTTON_SIZE = 14
+export const RADIOBUTTON_MARGIN = 8
+
+const RadioButton = ({disabled, label, onSelect, selected, style}: Props) => {
+  const clickableStyle = disabled ? {} : globalStyles.clickable
+
+  const StyledDiv = glamorous(glamorous.Div)(({border, disabled, selected}) => ({
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.desktop.js]: This should be pulled outside of RadioButton so its not remade everytime this renders

File: shared/common-adapters/radio-button.native.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+import ClickableBox from './clickable-box'
+import Text from './text'
+import type {Props} from './radio-button'
+import {globalStyles, globalColors} from '../styles'
+
+export const RADIOBUTTON_SIZE = 22
+export const RADIOBUTTON_MARGIN = 8
+
+const RadioButton = ({disabled, label, onSelect, selected, style}: Props) => {
+  const boxStyle = {
```
Comment (chrisnojima) [inline shared/common-adapters/radio-button.native.js]: should use glamorous or at least pull the static things into a const at the bottom and just define the dynamic stuff here like

```
const boxStyle = {
  ..._baseBaseStyle,
  opacity: disabled ? 0.4 : 1,
borderColor: // etc
}
```

Comment (cjb) [inline shared/common-adapters/radio-button.native.js]: @chrisnojima Thanks!  Both done, switched over to glamorous on native too.  How's it look now?

**Comment by keybase-ci-visdiff** [comment]
The commits 13d4f1e08f14...f54fe0dc36e0 introduce visual changes on linux. <details><summary>:mag_right: 4 new, 2 changed</summary>

 * Added: **Radio_Button-Normal_unchecked**  
   Radio_Button-Normal_unchecked rendered
 * Added: **Radio_Button-Normal_selected**  
   Radio_Button-Normal_selected rendered
 * Added: **Radio_Button-Disabled_unselected**  
   Radio_Button-Disabled_unselected rendered
 * Added: **Radio_Button-Disabled_selected**  
   Radio_Button-Disabled_selected rendered
 * Changed: **Notifications-Unsub_All** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503506674155-13d4f1e08f14-f54fe0dc36e0-linux/Notifications-Unsub_All-13d4f1e08f14.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503506674155-13d4f1e08f14-f54fe0dc36e0-linux/Notifications-Unsub_All-f54fe0dc36e0.png)  
   Notifications-Unsub_All comparison
 * Changed: **Notifications-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503506674155-13d4f1e08f14-f54fe0dc36e0-linux/Notifications-Normal-13d4f1e08f14.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503506674155-13d4f1e08f14-f54fe0dc36e0-linux/Notifications-Normal-f54fe0dc36e0.png)  
   Notifications-Normal comparison
</summary>

**Comment by cjb** [comment]
@chrisnojima Oops, of course -- done.

**Comment by keybase-ci-visdiff** [comment]
The commits 13d4f1e08f14...2a053f3b6246 introduce visual changes on linux. <details><summary>:mag_right: 4 new</summary>

 * Added: **Radio_Button-Normal_unchecked**  
   Radio_Button-Normal_unchecked rendered
 * Added: **Radio_Button-Normal_selected**  
   Radio_Button-Normal_selected rendered
 * Added: **Radio_Button-Disabled_unselected**  
   Radio_Button-Disabled_unselected rendered
 * Added: **Radio_Button-Disabled_selected**  
   Radio_Button-Disabled_selected rendered
</summary>

**Pull request merged**